docs/generatedcodeguide.rst

```diff
@@ -0,0 +1,277 @@
+    */
+    var num: int(32);
+    
+    /*
+      Used to store encoded byte stream of unknown fields encountered while parsing.
+      As per proto3 documentation, unknown fields should be preserved and appended
+      to the generated message byte stream.
+    */
+    var unknownFieldStream: bytes = "";
+    
+    /*
+      User exposed method for serializing data to protobuf wire format.
+      This is a wrapper method to the actual method, it is used for making a
+      binary writing channel out of the channel given as an input by the user.
+    */
+    proc serialize(ch) throws { ... }
+    
+    /*
+      Contains the actual implementation for serializing data. Calls the
+      `Append` functions of the user support library. It appends the `unknownFieldStream`
+      at the end of the message. This should end up as a private method when supported,
+      so a user should not call it directly.
+    */
+    proc _serialize(binCh) throws { ... }
+    
+    /*
+      User exposed method for parsing data from protobuf wire format.
+      This is a wrapper method to the actual method, it is used for making a
+      binary reading channel out of the channel given as an input by the user.
+    */
+    proc deserialize(ch) throws { ... }
+    
+    /*
+      Contains the actual implementation for parsing data. Calls the `Consume`
+      functions of the user support library. Appends unknown fields encountered
+      to the `unknownFieldStream` variable. This should end up as a private method
+      when supported, so a user should not call it directly.
+    */
+    proc _deserialize(binCh) throws { ... }
+  
+  }
+  
+Fields
+------
+The protocol buffer compiler generates a Chapel record field for each field defined
+within a message. Methods equivalent to `get` and `set` in other languages are
+implicitly generated by the Chapel compiler, so do not need to be generated by
+the protocol buffer compiler.
+
+Scalar Value Types
+^^^^^^^^^^^^^^^^^^
+A scalar message field can have one of the following types, the table shows the
+type specified in the ``.proto`` file, and the corresponding generated Chapel type:
+
+..
+  This table is intended to match the order given by the proto3 languge documentation.
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - .proto Type
+     - Chapel Type
+   * - double
+     - real(64)
+   * - float
+     - real(32)
+   * - int32
+     - int(32)
+   * - int64
+     - int(64)
+   * - uint32
+     - uint(32)
+   * - uint64
+     - uint(64)
+   * - sint32
+     - int(32)
+   * - sint64
+     - int(64)
+   * - fixed32
+     - uint(32)
+   * - fixed64
+     - uint(64)
+   * - sfixed32
+     - int(32)
+   * - sfixed64
+     - int(64)
+   * - bool
+     - bool
+   * - string
+     - string
+   * - bytes
+     - bytes
+
+Singular Fields
+^^^^^^^^^^^^^^^
+Every singular `message field`_ generates a record field variable of an appropriate Chapel type.
```

Comment:
I'd put the link on "singular" and "repeated", since those are the terms that I think users will be more confused about.  I believe it is possible to make rst links display differently but still refer to the same place without having to make a separate link at the bottom, but I don't remember the syntax off the top of my head.

Reply:
I am not able to find that syntax through googling :(